pkg/network/network.go

```diff
@@ -136,6 +140,75 @@ func Render(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, opensh
 	return objs, nil
 }
 
+// RenderObjsToRemove creates removal component object list
+func RenderObjsToRemove(prev, conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string, openshiftNetworkConfig *osv1.Network, clusterInfo *ClusterInfo) ([]*unstructured.Unstructured, error) {
+	log.Print("starting rendering objects to delete phase")
+	objsToRemove := []*unstructured.Unstructured{}
+	o := []*unstructured.Unstructured{}
+	var err error
+
+	operatorNamespace := os.Getenv("OPERATOR_NAMESPACE")
+
+	if conf.Multus == nil && prev != nil {
```

Comment:
conf.Multus == nil  must be checked, so a module wont find itself both in the create list and the remove list if we dont check it.

since renderMultus checks just for its conf (which is prev in our case) for 
conf.Module != nil and not conf != nil (s/conf/prev/), we are doing it here

Reply:
I see, thanks for the explanation

Reply:
btw, i would even move the conf != nil into the leaf functions, its a good practice that the function that use a parameter is the one that test it, up to you